src/quota.rs

```diff
@@ -0,0 +1,201 @@
+        for resource in quota_resources {
+            message.push_str(&match &resource.name {
+                Atom(name) => {
+                    format!("[{}/{}] {}\n", resource.usage, resource.limit, name)
+                }
+                Message => {
+                    format!(
+                        "{}/{} {}\n",
+                        resource.usage, resource.limit, messages_stock_string
+                    )
+                }
+                Storage => {
+                    let used = (resource.usage * 1024)
+                        .file_size(file_size_opts::BINARY)
+                        .map_err(|err| anyhow!("{}", err))?;
+                    let limit = (resource.limit * 1024)
+                        .file_size(file_size_opts::BINARY)
+                        .map_err(|err| anyhow!("{}", err))?;
+                    format!("{}/{} {}\n", used, limit, storage_stock_string)
+                }
+            });
+        }
+    }
+    Ok(message)
+}
+
+async fn get_unique_quota_roots_and_usage(
+    folders: Vec<String>,
+    imap: &mut Imap,
+) -> Result<Vec<(String, Vec<QuotaResource>)>> {
+    // IDEA: OPTIMIZATION:
+    // the unique quota roots of get_unique_quota_roots_and_usage could be cached and then the server could be asked for the quotas of those.
+    // this would reduce incoming traffic and for most cases also outgoing traffic, because most email server share quota roots across folders/mailboxes.
+    // Again at the cost of increasing code complexity and the question for how long it should be cached
+    // IDEA2: maybe the provider db could also contain quota root names?
+    let mut unique_quota_roots: Vec<(String, Vec<QuotaResource>)> = Vec::new();
+    for folder in folders {
+        let (quota_roots, quotas) = &imap.get_quota_roots(&folder).await?;
+        // if there are new quota roots found in this imap folder, add them to the list
+        for qr_entries in quota_roots {
+            for quota_root_name in &qr_entries.quota_root_names {
+                // the quota for that quota root
+                let quota: Quota = quotas
+                    .iter()
+                    .find(|q| &q.root_name == quota_root_name)
+                    .cloned()
+                    .ok_or_else(|| anyhow!("quota_root should have a quota"))?;
+                match unique_quota_roots
+                    .iter()
+                    .find_position(|(root_name, _)| root_name == quota_root_name)
+                {
+                    None => {
+                        unique_quota_roots.push((quota_root_name.clone(), quota.resources));
+                    }
+                    Some((position, ..)) => {
+                        // replace old quotas, because between fetching quotaroots for folders,
+                        // messages could be recieved and so the usage could have been changed
+                        unique_quota_roots.get_mut(position).unwrap().1 = quota.resources;
```

Comment:
Better to still use `if let` even though you are sure this `unwrap` will never fail. Because each time core crashes due to `unwrap` on a phone without debug info we have to go through all `unwrap`s and check them. Ideally we should forbid `unwrap`s with clippy at some point.